packages/block-editor/src/hooks/margin.js

```diff
@@ -159,6 +160,7 @@ export function MarginEdit( props ) {
 						units={ units }
 						allowReset={ false }
 						splitOnAxis={ false }
+						setMouseOver={ setMouseOver }
```

Comment:
For completeness and consistency, I think it would be good to pass `onMouseOver` and `onMouseOut` through the `BoxControl` so the visualizers when there aren't spacing presets behave the same way as with the newer control.

What do you think?

Reply:
Are you suggesting `onMouseOver, onMouseOut` instead of just passing in `setMouseOver` as it is a better public API for something in the components package? If we do that I wonder if we should use that API in the spacing presets component as well for consistency.

Reply:
I only had a quick glance at the BoxControl, but it looked like it passed extra props on through to the underlying UnitControl/InputControls. So I figured we could just pass the onMouseOver and onMouseOut to it so they get assigned to its UnitControl the same way you did within the SpacingInputControl.

The main issue was making sure that the new visualizer UX is there for non-spacing-preset controls as well.

Reply:
My original comment might not have been clear as it was attached to where you pass `setMouseOver` to the SpacingInputControl, sorry. It's [this control](https://github.com/WordPress/gutenberg/blob/a527db4b556812c17c8bbc55036129a2e759f718/packages/block-editor/src/hooks/margin.js#L144-L152) that I think would be good to display the visualizer when hovered.

Reply:
Gotcha, will take a closer look on Monday.

Reply:
@aaronrobertshaw passing the mouse callbacks to the top level of the BoxControl seems to work so have pushed this change.